Sources/MongoSwift/MongoCursor.swift

```diff
@@ -48,18 +51,8 @@ public class MongoCursor<T: Codable>: Sequence, IteratorProtocol {
 
     /// Cleans up internal state.
     deinit {
-        self.close()
-    }
-
-    /// Closes the cursor.
-    public func close() {
-        guard let cursor = self._cursor else {
-            return
-        }
-        mongoc_cursor_destroy(cursor)
-        self._cursor = nil
-        self._client = nil
-        self._session = nil
+        self._client.connectionPool.checkIn(self._connection)
```

Comment:
something worth considering in the future: we should return the connection as soon as the cursor is exhausted. Right now I could fully iterate a cursor, print out its contents, but accidentally keep some reference to it, and run the risk of exhausting our connection pool. At very least we should thoroughly document the interplay of connections and cursors.

Reply:
I agree we should probably just be returning the connections upon exhausting the cursor. It will require us to keep track of whether the cursor is exhausted internally, which is sorta equivalent to the cursor being closed early (which we were trying to avoid with these changes unfortunately)

Reply:
I agree , I've opened [SWIFT-517](https://jira.mongodb.org/browse/SWIFT-517) about doing this. 
there's some trickiness we need to handle with checking for errors after exhaustion, and also with knowing when the cursor is actually exhausted, since we don't currently track cursor type in `MongoCursor`. so I think we should do this shortly but I'm going to defer it to another ticket.